src/com/dis2/colorPicker/WheelColor.java

```diff
@@ -0,0 +1,198 @@
+        this.setSize(new Dimension(width, height));
+
+        this.addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+
+                if (isInsideCharecter(e.getPoint()) && !isInsideWheelColor(e.getPoint())) {
+                    setCharacterClicked(true);
+                    coordinates = e.getPoint();
+                } else if (isInsideWheelColor(e.getPoint())) {
+                    setCharacterClicked(true);
+                    updateSelectedCharacter(e.getPoint());
+                } else {
+                    setCharacterClicked(false);
+                }
+                repaint();
+            }
+
+            @Override
+            public void mousePressed(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseReleased(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseEntered(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+
+            }
+        });
+    }
+
+    public boolean isInsideCharecter(Point p) {
+        int xIni = getImagenCenterX(characters.get(selectedCharacter).getCharacter());
+        int yIni = getImagenCenterY(characters.get(selectedCharacter).getCharacter());
+        int xEnd = xIni + characters.get(selectedCharacter).getCharacterWidth();
+        int yEnd = yIni + characters.get(selectedCharacter).getCharacterHeight();
+
+        if (p.x >= xIni && p.x <= xEnd && p.y >= yIni && p.y <= yEnd) {
+            return true;
+        }
+        return false;
+    }
+
+    public void updateSelectedCharacter(Point p) {
+        int auxCharacter = selectedCharacter;
+        double angle = Math.atan2((double) (p.y - coordinates.y), (double) (p.x - coordinates.x));
+        angle = (angle > 0 ? angle : (2 * Math.PI + angle)) * 360 / (2 * Math.PI);
+        selectedCharacter = Math.abs(characters.size() - (int) Math.floor(angle / (360 / characters.size()))) - 1;
+        selectedCharacter = (selectedCharacter == -1) ? auxCharacter : selectedCharacter;
+        
+    }
+
+    public boolean isInsideWheelColor(Point p) {
+        if (!isCharacterClicked()) {
+            return false;
+        } 
+        int distancesquared = (p.x - coordinates.x) * (p.x - coordinates.x) + (p.y - coordinates.y) * (p.y - coordinates.y);
+        return distancesquared <= wheelRadius * wheelRadius;
+    }
+
+    public int getWheelRadius() {
+        return wheelRadius;
+    }
+
+    public void setWheelRadius(int wheelRadius) {
+        this.wheelRadius = wheelRadius;
+    } 
+
+    public void addCharacter(Character character) {
+        characters.add(character);
+    }
+
+    public int getImagenCenterX(Image character) {
+        return (this.getWidth() / 2) - (character.getWidth(this) / 2);
+    }
+
+    public int getImagenCenterY(Image character) {
+        return (this.getHeight() / 2) - (character.getHeight(this) / 2);
+    }
```

Comment:
These getImageCenterX and getImageCenterY can be use accross the app. Can you created a new "util.java" file inside the "Shared" folder, put the functions there and make them static so other classes can use them without instantiating new object of it